app/models/ability.rb

```diff
@@ -73,12 +73,15 @@ def signed_in(user)
 
     can [:create, :destroy], Subscription, user_id: user.id
 
-    can :manage, Event do |event|
+    can [:index, :show, :edit, :destroy, :update, :confirm, :restart], Event do |event|
       event.users.include?(user)
     end
-    # can create an event until the last day of a conference
-    can :create, Event, program_id: Conference.where('end_date >= ?', Date.today).map { |conference| conference.program.id}.compact
 
+    # can create an event until the last day of a conference
+    # can create an event if program has call for papers defined
+    can [:new, :create], Event do |event|
+      event.program.cfp_open? && event.new_record? && event.program_id == Conference.where('end_date >= ?', Date.today).map { |conference| conference.program.id}.compact
```

Comment:
I am not sure we need the event.program_id.
I know it comes from a model validation, but the model does not validate against cfp_open? and cfp is already before the end of the conference.